### PR TITLE
[CLI-195] Use graphql to list devices

### PR DESCRIPTION
**Pull request by codemodify**
### Description
use graphql to list devices

### Checklist
- [x] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [x] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [x] I have added one or more reviewers
- [x] This PR only contains one isolated change (bug fix, feature, etc)
- [x] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [x] I have notified reviewers on Slack by @ mentioning them in the `#cli`channel
- [x] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Ticket
https://remoteit.atlassian.net/browse/CLI-195

### CODE REVIEW

File: api/graphql-client.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	req, err := http.NewRequest("POST", thisRef.endpointURL, bytes.NewBuffer(data))
+	if err != nil {
+		logger.Errorf("GraphQLClient: error creating request for [%s], details: [%s]", thisRef.endpointURL, err)
+		return nil, err
+	}
+
+	req.Header.Add("token", thisRef.token)
+	resp, err := thisRef.client.Do(req)
+	if err != nil {
+		logger.Errorf("GraphQLClient: error sending request for [%s], details: [%s]", thisRef.endpointURL, err)
```
Comment (danawoodman) [inline api/graphql-client.go]: Stop using brackets please 😉

File: api/graphql-client.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	requestAsBytes, err := json.Marshal(request)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return thisRef.request(requestAsBytes)
+}
+
+// DeviceListAllGQL - lists all deivces for the user using GraphQL
+func (thisRef GraphQLClient) DeviceListAllGQL(uid string) ([]Device, error) {
```
Comment (danawoodman) [inline api/graphql-client.go]: If it is just to find one device, why call it `DeviceListAll`?

I propose calling it something like `FindDeviceByUID` or something similar, thoughts?

File: api/graphql-client.go (hunk 1 above)
Comment (codemodify) [reply api/graphql-client.go]: fixed

File: api/graphql-client.go (hunk 2 above)
Comment (codemodify) [reply api/graphql-client.go]: fixed

File: connections/localproxy/through-remoteit-proxy.go [hunk 3]
```diff
@@ -28,20 +28,28 @@ type localProxyToRemoteitProxy struct {
...
+	graphQLClient := api.NewGraphQLClient(token)
+	devices, err := graphQLClient.FindDeviceByUID(initiator.UID)
+	if err != nil {
+		logger.Error(err)
+		return NewLocalProxyToDeadEnd()
+	}
+	if len(devices) <= 0 {
+		logger.Errorf("FindDeviceByUID returned no devices for UID %s, details: %s", initiator.UID, err.Error())
+		return NewLocalProxyToDeadEnd()
 	}
```
Comment (danawoodman) [inline connections/localproxy/through-remoteit-proxy.go]:
```suggestion
	device, err := graphQLClient.FindServiceByUID(initiator.UID)
	if err != nil {
		logger.Error(err)
		return NewLocalProxyToDeadEnd()
	}
	if device != nil {
		logger.Errorf("FindDeviceByUID returned no device for UID %s, details: %s", initiator.UID, err.Error())
		return NewLocalProxyToDeadEnd()
	}
```

File: connections/localproxy/through-remoteit-proxy.go [hunk 4]
```diff
@@ -28,20 +28,28 @@ type localProxyToRemoteitProxy struct {
...
+	deviceType := devices[0].DeviceType
```
Comment (danawoodman) [inline connections/localproxy/through-remoteit-proxy.go]:
```suggestion
	deviceType := device.DeviceType
```

**Pull request merged**